File: pi.go
package main

import "fmt"

func pi_nks(limit int) float64 {
	pi := 3.0
	s := 1
	for i := 2; i <= limit; i += 2 {
		pi += (float64(s*4) / float64(i*(i+1)*(i+2)))
		s = s * (-1)
	}
	return pi
}

func pi_gls(limit int) float64 {
	pi := 0.0
	s := 1.0
	for i := 1; i <= limit; i += 2 {
		pi += float64(s*4) / float64(i)
		s = s * (-1)
	}
	return pi
}

func main() {
	const LIMIT = 100
	fmt.Printf("NKS: %.13f\n", pi_nks(LIMIT))
	fmt.Printf("GLS: %.13f\n", pi_gls(LIMIT))
}

File: pi.nim
import strformat

proc pi_nks(l: int): float=
  var pi: float = 3.0
  var s: int = 1
  for i in countup(2, l, 2):
    pi += s*4/(i*(i+1)*(i+2))
    s = s*(-1)
  pi

proc pi_gls(l: int): float=
  var pi: float = 0.0
  var s: int = 1
  for i in countup(1, l, 2):
    pi += float(s*4/i)
    s = s*(-1)
  pi

const limit: int = 100
echo fmt("NKS: {pi_nks(limit):.13f}")
echo fmt("GLS: {pi_gls(limit):.13f}")
